cmd/frontend/graphqlbackend/graphqlbackend.go

```diff
@@ -607,25 +410,88 @@ type schemaResolver struct {
-func newSchemaResolver(db dbutil.DB, repoupdaterClient *repoupdater.Client) *schemaResolver {
-	if repoupdaterClient == nil {
-		repoupdaterClient = repoupdater.DefaultClient
-	}
+// newSchemaResolver will return a new schemaResolver using repoupdater.DefaultClient.
+func newSchemaResolver(db dbutil.DB) *schemaResolver {
 
-	return &schemaResolver{
+	r := &schemaResolver{
 		db:                db,
-		repoupdaterClient: repoupdaterClient,
+		repoupdaterClient: repoupdater.DefaultClient,
 
-		BatchChangesResolver: defaultBatchChangesResolver{},
-		AuthzResolver:        defaultAuthzResolver{},
-		CodeIntelResolver:    defaultCodeIntelResolver{},
-		InsightsResolver:     defaultInsightsResolver{},
-		LicenseResolver:      defaultLicenseResolver{},
+		AuthzResolver:     defaultAuthzResolver{},
+		CodeIntelResolver: defaultCodeIntelResolver{},
+		InsightsResolver:  defaultInsightsResolver{},
+		LicenseResolver:   defaultLicenseResolver{},
 	}
+
+	r.nodeByIDFns = map[string]NodeByIDFunc{
+		"AccessToken": func(ctx context.Context, id graphql.ID) (Node, error) {
+			return accessTokenByID(ctx, db, id)
+		},
+		"ProductLicense": func(ctx context.Context, id graphql.ID) (Node, error) {
+			if f := ProductLicenseByID; f != nil {
+				return f(ctx, db, id)
+			}
+			return nil, errors.New("not implemented")
+		},
+		"ProductSubscription": func(ctx context.Context, id graphql.ID) (Node, error) {
+			if f := ProductSubscriptionByID; f != nil {
+				return f(ctx, db, id)
+			}
+			return nil, errors.New("not implemented")
+		},
+		"ExternalAccount": func(ctx context.Context, id graphql.ID) (Node, error) {
+			return externalAccountByID(ctx, db, id)
+		},
+		externalServiceIDKind: func(ctx context.Context, id graphql.ID) (Node, error) {
+			return externalServiceByID(ctx, db, id)
+		},
+		"GitRef": func(ctx context.Context, id graphql.ID) (Node, error) {
+			return r.gitRefByID(ctx, id)
+		},
+		"Repository": func(ctx context.Context, id graphql.ID) (Node, error) {
+			return r.repositoryByID(ctx, id)
+		},
+		"User": func(ctx context.Context, id graphql.ID) (Node, error) {
+			return UserByID(ctx, db, id)
+		},
+		"Org": func(ctx context.Context, id graphql.ID) (Node, error) {
+			return OrgByID(ctx, db, id)
+		},
+		"OrganizationInvitation": func(ctx context.Context, id graphql.ID) (Node, error) {
+			return orgInvitationByID(ctx, db, id)
+		},
+		"GitCommit": func(ctx context.Context, id graphql.ID) (Node, error) {
+			return r.gitCommitByID(ctx, id)
+		},
+		"RegistryExtension": func(ctx context.Context, id graphql.ID) (Node, error) {
+			return RegistryExtensionByID(ctx, db, id)
+		},
+		"SavedSearch": func(ctx context.Context, id graphql.ID) (Node, error) {
+			return r.savedSearchByID(ctx, id)
+		},
+		"Site": func(ctx context.Context, id graphql.ID) (Node, error) {
+			return r.siteByGQLID(ctx, id)
+		},
+		"LSIFUpload": func(ctx context.Context, id graphql.ID) (Node, error) {
+			return r.LSIFUploadByID(ctx, id)
+		},
```

Comment:
```suggestion
		"LSIFUpload": r.LSIFUploadByID,
```

I think if we do this we would encourage a more standard way to get data out of a db handle (this map contains several different ways to pass db - via `r`, via an explicit parameter, etc).

Reply:
It would require to change the return type of that function to `Node`, though. Maybe that's not so bad, though.

Reply:
Oh nvm I thought it had a less specific type.